prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 Red Hat Developer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.parodos.tasks.project.checker;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import com.redhat.parodos.project.enums.ProjectAccessStatus;
+import com.redhat.parodos.utils.RestUtils;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.task.checker.BaseWorkFlowCheckerTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import com.redhat.parodos.workflows.workflow.WorkFlow;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.http.ResponseEntity;
+
+import static com.redhat.parodos.tasks.project.consts.ProjectAccessRequestConstant.ACCESS_REQUEST_ID;
+
+/**
+ * Project access request approval workflow checker task
+ *
+ * @author Annel Ketcha (Github: anludke)
+ */
+
+@Slf4j
+public class ProjectAccessRequestApprovalWorkFlowCheckerTask extends BaseWorkFlowCheckerTask {
+
+	private final String serviceUrl;
+
+	private final String servicePort;
+
+	private final String serviceUsername;
+
+	private final String servicePassword;
+
+	public ProjectAccessRequestApprovalWorkFlowCheckerTask(WorkFlow projectAccessRequestApprovalEscalationWorkFlow,
+			long sla, String serviceUrl, String servicePort, String serviceUsername, String servicePassword) {
+		super(projectAccessRequestApprovalEscalationWorkFlow, sla);
+		this.serviceUrl = serviceUrl;
+		this.servicePort = servicePort;
+		this.serviceUsername = serviceUsername;
+		this.servicePassword = servicePassword;
+	}
+
+	@Override
+	public WorkReport checkWorkFlowStatus(WorkContext workContext) {
+		log.info("Start ProjectAccessRequestApprovalWorkFlowCheckerTask...");
+		UUID accessRequestId;
+		try {
+			accessRequestId = UUID.fromString(getRequiredParameterValue(ACCESS_REQUEST_ID));
+		}
+		catch (MissingParameterException e) {
+			log.error("Exception when trying to get required parameter(s): {}", e.getMessage());
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		try {
+			String url = String.format("http://%s:%s/api/v1/projects/access/%s", serviceUrl, servicePort,
+					accessRequestId);
```

Comment:
What is the service/server that should reply to `"http://%s:%s/api/v1/projects/%s/access` ?

Reply:
So this is the new project access endpoint:
```
@RequestMapping("/api/v1/projects/access")
```
why not use rest call? shouldn't we use some service API call or is the call done via HTTP to remain detached from the service?

Reply:
> What is the service/server that should reply to `"http://%s:%s/api/v1/projects/%s/access` ?

The task uses the notification service to send this approval url to the approver.

Reply:
> 
Actually, the url is the endpoint path to make a POST HTTP call to approve the request. Ideally there will be a UI with the form to approve/reject the request with a comment.
The task showcases how the endpoint and required params (access request id) will send out to the approver via notification service.

Reply:
I'd rather have this implemented in a similar fashion as https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/main/java/com/redhat/parodos/NotificationSender.java#L21
Leveraging the common SDK to interact with the service endpoint.

Reply:
I was debating on using the SDK but @RichardW98 convinced me :)
I'm happy to refactor this in a separate PR. I'll create a tech debt to track and address this.
Thanks